packages/core/core/src/requests/AssetGraphRequest.js

```diff
@@ -130,6 +137,7 @@ export class AssetGraphBuilder {
     this.name = name;
     this.requestedAssetIds = requestedAssetIds ?? new Set();
     this.shouldBuildLazily = shouldBuildLazily ?? false;
+    this.shouldBundle = false;
```

Comment:
This may require more thought, but essentially keeping a higher level identifier if the asset graph request will need to bundle or not. It's set to true if it meets certain criteria.